src/App.tsx

```diff
@@ -9,52 +13,102 @@ import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [currUser, setCurrUser] = useState<User | null>(null);
+  const [currPost, setCurrPost] = useState<Post | null>(null);
+  const [openModal, setOpenModal] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  const searchUserPosts = (userId: number) => {
+    getUserPosts(userId)
+      .then(setPosts)
+      .catch(() => setError(true))
+      .finally(() => setLoading(false));
```

Comment:
The error handling in the `searchUserPosts` function is quite generic. Consider providing more informative feedback or logging for debugging purposes. This will help in understanding what went wrong during the API call.